pkg/addons/helpers.go

```diff
@@ -42,3 +46,29 @@ func vSphereService() *corev1.Service {
 		},
 	}
 }
+
+// AreEmbeddedAddons checks if all specified addons are embedded addons
+func AreEmbeddedAddons(addons []kubeoneapi.Addon) (bool, error) {
+	// Read the directory entries for embedded addons
+	embeddedAddons, err := fs.ReadDir(embeddedaddons.FS, ".")
+	if err != nil {
+		return false, err
+	}
+
+	// Iterate over addons specified in the KubeOneCluster object
+	for _, addon := range addons {
+		embedded := false
+		// Iterate over embedded addons directory to check if the addon exists
+		for _, embeddedAddon := range embeddedAddons {
+			if embeddedAddon.Name() == addon.Name {
+				embedded = true
+				continue
```

Comment:
Shouldn't this be `break` instead?

```suggestion
				break
```